jwst/ramp_fitting/ramp_fit.py

```diff
@@ -383,87 +417,77 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
             var_p4[num_int, :, rlo:rhi, :] = den_p3 * med_rates[ rlo:rhi, :]
             var_r4[num_int, :, rlo:rhi, :] = num_r3 * den_r3
 
-        var_p4[var_p4 < 0.] = 0. # ignore contributions from negative slopes
-        var_p4 = utils.rem_nan_inf(var_p4)
-
-        var_r4[var_r4 < 0.] = 0.
-        var_r4 = utils.rem_nan_inf(var_r4)
-
-        inv_var_p4[num_int, :, :, :] = 1./var_p4[num_int, :, :, :]
-        inv_var_p4 = utils.rem_nan_inf(inv_var_p4)
-        s_inv_var_p3[num_int, :, :] = (inv_var_p4[num_int, :, :, :]).sum(axis=0)
-
-        var_p3[num_int, :, :] = 1./ s_inv_var_p3[num_int, :, :]
-        var_p3 = utils.rem_nan_inf(var_p3)
-
-        inv_var_r4[num_int, :, :, :] = 1./var_r4[num_int, :, :, :]
-        inv_var_r4 = utils.rem_nan_inf(inv_var_r4)
-
-        s_inv_var_r3[num_int, :, :] = (inv_var_r4[num_int, :, :, :]).sum(axis=0)
-        var_r3[num_int, :, :] = 1./ s_inv_var_r3[num_int, :, :]
-        var_r3 = utils.rem_nan_inf(var_r3)
+            del den_r3, den_p3, num_r3, segs_beg_3
+            del gain_sect
+            del gdq_sect
+
+        # The next 4 statements zero out entries for non-existing segments, and
+        #   set the variances for segments having negative slopes (the segment
+        #   variance is proportional to the median estimated slope) to
+        #   outrageously large values so that they will have negligible
+        #   contributions.
+        var_p4[num_int,:,:,:] *= ( segs_4[num_int,:,:,:] > 0)
+        var_p4[var_p4 <= 0.] = utils.LARGE_VARIANCE
+        var_r4[num_int,:,:,:] *= ( segs_4[num_int,:,:,:] > 0)
+        var_r4[var_r4 <= 0.] = utils.LARGE_VARIANCE
+
+        s_inv_var_p3[num_int, :, :] = (1./var_p4[num_int, :, :, :]).sum(axis=0)
+        var_p3[num_int, :, :] = 1./ (1./var_p4[num_int, :, :, :]).sum(axis=0)
```

Comment:
Is there a reason for recomputing the sum?  It looks as if you could use `s_inv_var_p3`.